packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3199,10 +3199,14 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     RenderObject? result;
     void visit(Element element) {
       assert(result == null); // this verifies that there's only one child
-      if (element is RenderObjectElement)
-        result = element.renderObject;
-      else
+      if (element is RenderObjectElement) {
+        // Don't use the getter, the _renderObject may be null if the element
+        // is defunct, and the getter asserts that it is not null. This method
+        // does not guarantee a non-null return anyway.
+        result = element._renderObject;
```

Comment:
can we instead return null immediately at the top of this method if the element is defunct?
Seems like a bug if you ever had non-defunct elements in a tree with defunct elements.

Reply:
Done. We have other assertions around element that make sure you don't end up in that kind of state.